### PR TITLE
non-major features

### LINKED ISSUE #411: transactionDictionary doesn't support register entity
https://github.com/oasisprotocol/oasis-wallet-web/blob/7094119a10c9bbc1119a371cd2110d9ef2127630/src/app/components/Transaction/index.tsx#L159

if we encounter a type that's not supported, this line crashes. would be good if we could do something graceful and continue processing other transactions

### PR DESCRIPTION
**Pull request by pro-wh**
- a new codepath for #411. no automated tests. missing french translation
- failed block explorer requests were unhelpfully showing up as "undefined," an improvement to that

fixes #411

part of #403

### CODE REVIEW

**Comment by pro-wh** [comment]
image

oh and I didn't write a test for this new part

**Comment by pro-wh** [comment]
adding test

File: README.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Oasis Wallet
 
 [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
-[![codecov](https://codecov.io/gh/esya/oasis-wallet/branch/develop/graph/badge.svg)](https://codecov.io/gh/esya/oasis-wallet)
+[![codecov](https://codecov.io/gh/oasisprotocol/oasis-wallet/branch/develop/graph/badge.svg)](https://codecov.io/gh/oasisprotocol/oasis-wallet)
 [![Renovate enabled](https://img.shields.io/badge/renovate-enabled-brightgreen.svg)](https://renovatebot.com/)
-[![Build status](https://github.com/esya/oasis-wallet/actions/workflows/build-test.yaml/badge.svg)](https://github.com/esya/oasis-wallet/actions)
+[![Build status](https://github.com/oasisprotocol/oasis-wallet/actions/workflows/build-test.yaml/badge.svg)](https://github.com/oasisprotocol/oasis-wallet/actions)
```
Comment (lukaw3d) [inline README.md:6]: oasisprotocol/oasis-wallet-web*

Comment (pro-wh) [reply README.md:6]: oh yeah thanks

File: src/app/components/Transaction/index.tsx [hunk 2]
```diff
@@ -156,10 +157,24 @@ export function Transaction(props: TransactionProps) {
     },
   }
 
-  const matchingConfiguration = transactionDictionnary[transaction.type as TransactionType][side]
-  const icon = matchingConfiguration.icon()
-  const header = matchingConfiguration.header()
-  const designation = matchingConfiguration.designation
+  let icon, header, designation
```
Comment (lukaw3d) [inline src/app/components/Transaction/index.tsx:160]: These then have type `any` :/

Alternative:
```ts
  const unrecognizedTransaction: TransactionDictionary[TransactionType][TransactionSide] = {
    designation: t('account.otherTransaction.designation', 'Other address'),
    icon: () => <New />,
    header: () => (
      <Trans
        i18nKey="account.otherTransaction.header"
        t={t}
        components={[transaction.type]}
        defaults="Unrecognized transaction, type '<0></0>'"
      />
    ),
  }

  const isTypeRecognized = (type: string | undefined): type is TransactionType =>
    type ? type in transactionDictionary : false

  const matchingConfiguration = isTypeRecognized(transaction.type)
    ? transactionDictionary[transaction.type][side]
    : unrecognizedTransaction

  const icon = matchingConfiguration.icon()
  const header = matchingConfiguration.header()
  const designation = matchingConfiguration.designation
```

**Comment by pro-wh** [comment]
links fixed, rebased, regenerated snapshot for the cursor change

File: src/app/components/Transaction/index.tsx (hunk 2 above)
Comment (pro-wh) [reply src/app/components/Transaction/index.tsx:161]: hm, `isTypeRecognized`

Comment (pro-wh) [reply src/app/components/Transaction/index.tsx:161]: thanks, I'm fine with that. I'll add this in a commit

**Comment by pro-wh** [comment]
incorporated alternative unknown transaction selector code

**Review by lukaw3d — APPROVED** [review]

**Pull request merged**